### ISSUE TITLE
Error when filename is '='

### ISSUE THREAD

**Issue by mjrw** [airblade/vim-gitgutter#307]
I have a file that is named `=` (a single equals sign). When opening this file I get this error:

    "=" 2L, 31C                                                                                                                                                                                                          
    Error detected while processing function gitgutter#process_buffer[19]..gitgutter#sign#update_signs[1]..gitgutter#sign#find_current_signs:                                                                            
    line   14:                                                                                                                                                                                                           
    E684: list index out of range: 1                                                                                                                                                                                     
    E15: Invalid expression: split(components[2], '=')[1]                                                                                                                                                                
    Press ENTER or type command to continue 

It seems that #273 addressed this for filenames *containing* `=` signs, but this bug still occurs when the filename is *exactly* `=`.

(Why do I have a file with such a silly name, you ask? It's a script for doing quick calculations on the command line by saying e.g. `= 2 + 2` instead of `echo $(( 2 + 2 ))`)

**Comment by airblade** [comment]
That's odd because the fix in #273 should also work in this case.  Specifically the `[2:]` in [this line](https://github.com/airblade/vim-gitgutter/blob/8b046bd5211d12629504cbd336e1ba350ebea54a/autoload/gitgutter/sign.vim#L80) should skip over the buffer name at the top of the list of signs.

Would you be able to add this snippet immediately above that line?

```viml
for line in split(signs, '\n')
  echom line
endfor
```

And then let me know what you get with `:mess`?  I would expect it to be something like:

```
--- Signs ---
Signs for <FILE>:
  line=42  id=3001  name=GitGutterLineAdded
  ...
```

But for you to get the error you're getting, the `Signs for <FILE>` line must be the second (or later) line, counting from zero, in the output...

**Comment by mjrw** [comment]
Oops, you're right, #273 did fix it, I was just accidentally using an older version of gitgutter. Sorry for the confusion!
